### PR TITLE
[nlohmann_json] Fix build on Windows by providing all necessary settings

### PR DESCRIPTION
**Pull request by Minimonium**
Specify library name and version:  **nlohmann_json/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
@Minimonium I'm curious if you are using taocpp, because there is a slack for that project under cppalliance Slack, but it's private, because only admin slackers can create public channels. If you are interested, I can send you an invite.

**Comment by Minimonium** [comment]
@uilianries I just checked every single project that uses CMake and looks like a header-only project, so taocpp projects were on the ride. 😄 But I'm always happy to join channels, you know. 😉

File: recipes/nlohmann_json/all/conanfile.py [hunk 1]
```diff
@@ -38,11 +40,15 @@ def build(self):
...
         cmake = self._configure_cmake()
         cmake.install()
         tools.rmdir(os.path.join(self.package_folder, "lib"))
         try:
             os.remove(os.path.join(self.package_folder, "nlohmann_json.natvis"))
         except FileNotFoundError:
             pass
+
+    def package_id(self):
+        self.info.header_only()
```
Comment (SpaceIm) [inline recipes/nlohmann_json/all/conanfile.py:54]:
```suggestion
    def package_id(self):
        self.info.settings.clear()
```

`multiple_headers` option produces two different packages I guess

Comment (Minimonium) [reply recipes/nlohmann_json/all/conanfile.py:54]: Good catch!

**Comment by Minimonium** [comment]
It seems that hooks don't handle well the case when a header-only package produces different package layouts for different options.

**Comment by jgsogo** [comment]
I think that the problem with the hook is related to this `_is_header_only` implementation [here](https://github.com/conan-io/hooks/blob/a83ed36c9cc5d122072b56f2ae972975c072084f/hooks/conan-center.py#L798):

```python
def _is_recipe_header_only(conanfile):
    without_settings = not bool(_get_settings(conanfile))
    package_id_method = getattr(conanfile, "package_id")
    header_only_id = "self.info.header_only()" in inspect.getsource(package_id_method)
    return header_only_id or without_settings
```

This recipe is a header-only but it has different options that create different packages. That's totally ok. I've created a PR to the hooks: https://github.com/conan-io/hooks/pull/207

**Pull request merged**